src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderNamespaceTest.java

```diff
@@ -0,0 +1,153 @@
+    Mockito.when(manifestFile.read())
+        .thenReturn(
+            new ByteArrayInputStream(
+                String.join(
+                        "\n",
+                        "apiVersion: apps/v1",
+                        "kind: Deployment",
+                        "metadata:",
+                        "  name: nginx-deployment",
+                        "  labels:",
+                        "     app: nginx")
+                    .getBytes()));
+    Mockito.doAnswer(
+            invocation -> {
+              Yaml yaml = new Yaml();
+              Manifests.ManifestObject manifest =
+                  new Manifests.ManifestObject(
+                      yaml.load((String) invocation.getArguments()[0]), manifestFile);
+              Optional<String> namespace = manifest.getNamespace();
+              assertNotNull(namespace);
+              assertTrue(namespace.isPresent());
+              assertEquals("default", namespace.get());
+              return null;
+            })
+        .when(manifestFile)
+        .write(anyString(), anyString());
+    String commandNameSpace = KubernetesEngineBuilder.addNamespace(manifestFile, "*");
+    assertEquals("default", commandNameSpace);
+  }
+
+  @Test
+  public void testAddStarNamespaceWithExistingNamespace() throws IOException, InterruptedException {
+    FilePath manifestFile = Mockito.mock(FilePath.class);
+    Mockito.when(manifestFile.read())
+        .thenReturn(
+            new ByteArrayInputStream(
+                String.join(
+                        "\n",
+                        "apiVersion: apps/v1",
+                        "kind: Deployment",
+                        "metadata:",
+                        "  name: nginx-deployment",
+                        "  namespace: test",
+                        "  labels:",
+                        "     app: nginx")
+                    .getBytes()));
+    Mockito.doAnswer(
+            invocation -> {
+              Yaml yaml = new Yaml();
+              Manifests.ManifestObject manifest =
+                  new Manifests.ManifestObject(
+                      yaml.load((String) invocation.getArguments()[0]), manifestFile);
+              Optional<String> namespace = manifest.getNamespace();
+              assertNotNull(namespace);
+              assertTrue(namespace.isPresent());
+              assertEquals("test", namespace.get());
+              return null;
+            })
+        .when(manifestFile)
+        .write(anyString(), anyString());
+    String commandNameSpace = KubernetesEngineBuilder.addNamespace(manifestFile, "*");
+    assertEquals("test", commandNameSpace);
+  }
+
+  @Test
+  public void testAddCustomNamespaceWithNoPreviousNamespace()
+      throws IOException, InterruptedException {
+    FilePath manifestFile = Mockito.mock(FilePath.class);
+    Mockito.when(manifestFile.read())
+        .thenReturn(
+            new ByteArrayInputStream(
+                String.join(
```

Comment:
Put this in a helper since it's been used a few times?

Reply:
Done.